### PR TITLE
Forensic mode v1.1

### PR DESCRIPTION
**Pull request by regit**
Update of #4372 

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Fix an issue in alert logging

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/509
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/285

### CODE REVIEW

File: src/output-json.c [hunk 1]
```diff
@@ -387,12 +391,115 @@ static void JsonAddMetadata(const Packet *p, const Flow *f, json_t *js)
...
+
+static void JsonAddStreamData(const Packet *p, const Flow *f, json_t *js,
+                              uint8_t flags, MemBuffer *payload)
+{
+    json_t *stream_js = json_object();
+    json_t *sjs = json_object();
+    json_t *cjs = json_object();
+
+    if (!stream_js || !cjs || !sjs)
+        return;
```
Comment (jlucovsky) [inline src/output-json.c]: Shouldn't the non-null items be called with `json_decref`?

Comment (regit) [reply src/output-json.c]: yes, looks like a good fix.

**Comment by regit** [comment]
resubmitting with fix following remark.

**Comment by pevma** [comment]
@regit  - I am testing out the next version here - https://github.com/regit/suricata/tree/forensic-mode-v2 and will report back.

**Pull request closed without merging**